x/fee_grant/internal/ante/fee.go

```diff
@@ -0,0 +1,94 @@
+package ante
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	// we depend on the auth module internals... maybe some more of this can be exported?
```

Comment:
Probably not.
If I understand correctly, usually I should only import `x/foo/exported` from another module, never from the other packages. This is impossible for 3 reasons:

1. We want to reuse the code in `authAnte.DeductFees`, although I could probably cut and paste it
2. `x/auth/keeper.AccountKeeper` is only available in that package, there is no available interface in `exported`
3. `x/auth/types.SupplyKeeper` is an external interface and could likely be copied into this module, however, `x/auth/types.FeeCollectorName` is a needed const to send fees and only available by importing that package

I am okay leaving these dependencies, as this is effectively an extension to the auth module that will only ever work with this module. However, I wanted to make this explicit, and if it is important to remove that dependency, I can do so.

Avoiding exposing those imports is why `internal/ante` is not in `alias.go` and `internal/types/tx` is a separate directory. If it is fine to make them a strong dependency, I will remove this comment and simplify the code and aliases

Reply:
> 2. `x/auth/keeper.AccountKeeper` is only available in that package, there is no available interface in exported

Every module should declare an expected_keeper with an interface of the used keepers and the functions used from them.

> 3. `x/auth/types.FeeCollectorName` is a needed const to send fees and only available by importing that package

Can you use the alias exposed for it (i.e just use `auth.FeeCollectorName`)?

Reply:
> Every module should declare an expected_keeper with an interface of the required keepers and the functions used from them.

It is not done in auth, and I am very hesitant to modify that core package, as it will lead to at least 4 review cycles. Maybe this can just be marked for a future cleanup issue if you think it is good to clean up?

> Can you use the alias exposed for it (i.e just use auth.FeeCollectorName)?

Importing `auth` is worse than importing the sub-packages as it pulls in even more (including the client logic). I was trying to reduce imports. I wanted to reduce it to the minimal sub-packages.

If I define some interfaces inside `fee_grant` and copy the `DeductFee` function, I could reduce imports to just `x/auth/types`. I didn't copy it, as @bez already complained about any code duplication (eg. StdTx -> GrantTx)